sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1194,17 +1196,27 @@ _pi_queue::_pi_queue(std::vector<ze_command_queue_handle_t> &ComputeQueues,
       ComputeQueueGroup.LowerIndex = FilterLowerIndex;
       ComputeQueueGroup.UpperIndex = FilterUpperIndex;
       ComputeQueueGroup.NextIndex = ComputeQueueGroup.LowerIndex;
-      // Create space to hold immediate commandlists corresponding to the
-      // ZeQueues
-      if (Device->useImmediateCommandLists()) {
-        ComputeQueueGroup.ImmCmdLists = std::vector<pi_command_list_ptr_t>(
-            ComputeQueueGroup.ZeQueues.size(), CommandListMap.end());
-      }
     } else {
       die("No compute queue available/allowed.");
     }
   }
 
+  if (Device->useImmediateCommandLists()) {
+    auto &ComputeQueueGroupRef = ComputeQueueGroup;
+    bool ThreadSpecificImmCmdLists = Device->useImmediateCommandLists() == 2;
```

Comment:
Please not use hardcoded literal constants in the code, use enums

Reply:
OK, done.